content/c-sharp/concepts/strings/terms/format/format.md

```diff
@@ -0,0 +1,237 @@
+    string formatString = "Hi {0}, your last login was on {1:d} at {1:t}.";
+
+ // Displaying a message using the current culture:
+
+    string message = String.Format(formatString, name, loginDate);
+    Console.WriteLine(message);          
+
+ // Displaying the same message but using an Italian culture:
+
+    CultureInfo culture = CultureInfo.CreateSpecificCulture("en-GB");
+    message = String.Format(culture, formatString, name, loginDate);
+
+    Console.WriteLine(message);
+ }
+}
+```
+
+The previous example outputs:
+
+```shell
+Hi John, your last login was on 9/2/24 at 5:45 PM.
+Hi John, your last login was on 02/09/2024 at 17:45.
+```
+
+## Codebyte Example
+
+The following runnable example defines a `User` class to populate a list of `User` objects and display them on the console. The user class uses the `.Format()` method to define a property (`FullName`) that returns a combination of two other fields (`firstName` and `lastName`). The `.Format()` method is also used to create the rows of user data to be displayed. Custom format specifiers are used to format the `Rank` (numeric) and `LastLogin` (date) properties of each user. The default format specifier `G` is used to format the enum property `Type`, since no format specifier is indicated.
+
+```codebyte/csharp
+using System;
+
+public enum UserType {None, Student, Teacher, Parent};
+
+public class User
+{
+   private readonly string firstName;
+   private readonly string lastName;
+   private readonly UserType type;
+   private readonly float rank;
+   private readonly DateTime lastLogin;
+
+   public User (string firstName, string lastName, UserType type, float rank, DateTime lastLogin)
+   {
+     this.firstName = firstName;
+     this.lastName = lastName;
+     this.type = type;
+     this.rank = rank;
+     this.lastLogin = lastLogin;
+   }
+
+   public string FirstName { get { return firstName; } }
+   public string LastName { get { return lastName; } }
+   public string FullName { get { return String.Format("{0}, {1}", lastName, firstName); } }
+   public UserType Type { get { return type; } }
+   public float Rank { get { return rank; } }
+   public DateTime LastLogin { get { return lastLogin; } }
+}
+
+public class Example {
+
+ static User[] UserList = new User[]
+ {
+   new User("Alice", "Apple", UserType.Student, 3.456f, new DateTime(2024, 9, 5, 14, 5, 34)),
+   new User("Robert", "Blue", UserType.Teacher, 2.453f, new DateTime(2024, 10, 15, 10, 20, 48)),
+   new User("John", "Green", UserType.Student, 4.63f, new DateTime(2024, 10, 31, 9, 5, 6)),
+   new User("Amelia", "Orange", UserType.Parent, 1.4f, new DateTime(2024, 11, 3, 17, 10, 24))
+ };
+
+ static void Main()
+ {
+    foreach(User user in UserList)
+    {
+      string line = String.Format("{0} ({1}). Rank: {2:0.00}. Last login on {3:MMM d, yyyy h:mm:ss tt}.",
+      user.FullName, user.Type, user.Rank, user.LastLogin);
+      Console.WriteLine(line);
+    }
+  }
+}
```

Comment:
This codebyte example can be a bit difficult for beginners as it involves a lot of new concepts such as classes and loops. We need to simplify this to something very basic where the reader has to only focus on the `.Format()` method and its results. 

Please consider something simple like this - 

```cs
using System;

class Program
{
    static void Main()
    {
        string firstName = "John";
        string lastName = "Doe";
        int age = 30;
        double height = 5.9;

        // Using String.Format to create a formatted string
        string formattedString = String.Format("Name: {0} {1}, Age: {2}, Height: {3} feet", firstName, lastName, age, height);

        Console.WriteLine(formattedString);
    }
}
```